### PR TITLE
Let 48 hour editing window of a flight depend on creation time

### PR DESCRIPTION
**Pull request by plaursen**
…, not flight time to allow for creation and editing of 'historical' flights.

Not the cleanest impl, but wanted to avoid adding a Flight.Created field, as the would have to be reconstructed from change history somewhere.

### CODE REVIEW

File: FlightJournal.Web/Controllers/FlightController.cs [hunk 1]
```diff
@@ -683,5 +666,16 @@ internal IEnumerable<int> GetLocationsAffectedByFlight(Flight flight)
             var affectedLocations = new[] { ClubController.CurrentClub.LocationId, flight.StartedFromId, flight.LandedOnId ?? 0, pilotClubLocationId, backSeatPilotClubLocationId, payerClubLocationId }.Distinct();
             return affectedLocations;
         }
+
+        public static bool UserCanEditFlight(FlightContext db, Guid flightId, IPrincipal user)
+        {
+            if (user.IsManager()) return true;
```
Comment (janhebnes) [inline FlightJournal.Web/Controllers/FlightController.cs:672]: IsEditor already incorporates the check for manager and administrator. 

Could you confirm if we have a retained that manager and editor is club specific, so the flight must be related to the club with pilot or location before they can edit it ?

File: FlightJournal.Web/Views/Report/Index.cshtml [hunk 2]
```diff
@@ -259,12 +259,9 @@
                             </td>
                         }
                         <td class="hidden-print text-right">
-                            @if (Request.IsPilot() && item.IsCurrentClubPilots())
+                            @if (Request.IsPilot() && item.IsCurrentClubPilots() && ((IEnumerable<Guid>)ViewBag.EditableFlights).Contains(item.FlightId))
                             {
-                                if (User.IsEditor() || item.Date.AddDays(3) >= DateTime.Now)
```
Comment (janhebnes) [inline FlightJournal.Web/Views/Report/Index.cshtml:264]: `|| item.Date.AddDays(3) >= DateTime.Now`
is present in Flight/PartialGrid.cshtml and in Flight/Index.cshtml

And we also have the comment on the page regarding how long a flight can be edited that might require editing.

**Comment by janhebnes** [comment]
Moving the editable flights from 48 hours after the day to depend on creation time means that we don't fully know at any point when a flight day can be regarded as "complete".

Today the logic in general is clear that a pilot is to make sure the list is complete and after midnight on day x editing is not longer possible and the list is to be regarded as "official". 

The general move to 48 hours after creation _could mean_ we should have an element of "official" when all flights are non-editable to general pilot, and we might also want to show creation time on the ones created "late". 

Depending on the motivation for this change I would propose that we consider extending editing up to a 4 or 5 days after, or alternatively keep the general 48 hours on a flight day, + extend with the proposed change in model where just created flights can be edited x hours after creation not matter at what time they where created.

**Comment by plaursen** [comment]
I hadn't thought of the need to consider a flight 'complete'. Technically, it never is (today), as Editors and above can (?) edit anyway.

The motivation was to facilitate addition of 'historical' flights, as was done to transfer FlightLogger status  (create a flight for each student on 20210101, with gradings reflecting FL status). Any mistankes done here require someone with elevated rights to edit.

Maybe to small a case for extending the editing period.
Let's leave this for further thought.

**Comment by janhebnes** [comment]
Det var oprindeligt tænkt ud fra klub regnskab hensyn og at en text logbog skal være skrevet når flyvedagen er ovre.

Har tænkt lidt mere over den og det at gamle flyvninger ikke kan redigeres lige efter de er oprettet tror jeg er en ret stor problemstilling i den nye måde det bruges på.

En fordel ved 48 timer på create er at den er nem at forstå. 

Hvad med kombi hvor "klubben"s piloter kan redigér med de nuværende spille regler, mens create tidspunktet bruges på de som er involveret i flyvningen pilot 1 og 2 og evt de som er i Change history?

**Comment by plaursen** [comment]
Det er ikke super akut, vi kan altid hjælpe dem som har behovet manuelt.

Og ved nærmere eftertanke er det nok renere bare at tilføje et 'Created' felt til Flight. Vi kan evt konstruere det ud fra change history under migration. Eller lade det være null, det burde jo egentlig kun være relevant fremadrettet ?

**Comment by plaursen** [comment]
Lukker den jher jf kommentarer ovenfor

**Pull request closed without merging**